notebook2/notebook/notebook.py

```diff
@@ -106,6 +107,24 @@ def get_name(line):
         "containing the name of the docker image to use for worker pods.") from e
 
 
+def requires_auth(for_page = True):
+    def auth(f):
+        @wraps(f)
+        def decorated(*args, **kwargs):
+            if 'user' not in session:
+                # Redirect to Login page here
```

Comment:
I think this is extremely clear from the `redirect(flask.url_for('login_page'))`, no?

Reply:
Yeah that shouldn't be there